### DISCUSSION TITLE
Request for tracking Scheduled / Programmed Maintenance Activities

### DISCUSSION THREAD

**Discussion by MenooMeow** [hargata/lubelog#774] [Ideas]
I've seen a few similar requests.  For example, to track snow tire installation.  I think what's really needed is a Scheduled Maintenance tracker.  The back of owner manuals have maintenance schedules.  It would be great to have a section where you can define any kind of scheduled/programmed maintenance task such as, but not limited to:

- Changing spark plugs
- Rotate tires
- Change oil
- Change timing belt
- Change water pump
- Replace valve cover gasket + spark plug tubes + rocker arm seals
- etc.

Each user defined activity would be triggered by either:
- Only a mileage trigger
- Only a time trigger
- Either a mileage or time trigger, whichever is first.

A key point here is that the schedule is "programmed" on a recurring basis where each instance is an event that must be accomplished.  For example, if changing the spark plugs is done every 40,000 miles, then it must be done at 40,000, 80,000, 120,000, 160,000, 200,000, 240,000, etc.  If I miss the 80,000 one, then it flags that event status as "missed" (if not accomplished at all) or "delayed" (if accomplished at any point in the future).  And, if the 80,000 mile event is done at 90,000, then the schedule is either re-calculated.  If re-calculated, then the next event is automatically reflowed to 90,000 (delayed service event) + 40,000 (activity default interval).  The idea is to be able to see whether maintenance tasks have been done at the regularly defined intervals or sporadically.  

Here's a partial screenshot of activities that I track on a spreadsheet by either mileage, time, or both.  

![mx schedule](https://github.com/user-attachments/assets/5e019847-aad5-42ee-9032-7f1945714678)

Oh, and the Dashboard initial page should be configurable to provide a "stop light" status display of all these maintenance activities.  

I see that reminders can be used to somewhat accomplish this but it's not a holistic solution.  I'm new to this incredible software so maybe it can already do it.  But if it can't, this would let me finally drop the spreadsheet :-)

Thank you!

**Comment by hargata** [comment]
This is literally the reminders feature.

**Reply by MenooMeow** [reply]
Reminders don't keep a history of a maintenance event against whether it was done or not.  It's literally just a reminder.  It doesn't do all the other things I said.  The reminder function is only a part of what I said.  If reminders did what I mentioned, I wouldn't be using a spreadsheet.

**Reply by hargata** [reply]
The reminders ties in to plan/service/repair/upgrade records. 

The reminders shouldn’t be used to track the history of a service, that’s what the service record tab is for.

when you create a service record, you can link a recurring reminder to it so that it gets renewed.

There are no plans for keeping track of how overdue a reminder is before it gets renewed, it only benefits a very small group of users and takes more effort than it’s worth.

**Reply by MenooMeow** [reply]
> The reminders ties in to plan/service/repair/upgrade records.
> 
> The reminders shouldn’t be used to track the history of a service, that’s what the service record tab is for.

The two should be inextricably linked.  A reminder is just a notification function, not an accounting function.  But the core necessity is accounting.  If the maintenance schedule is not followed, there has to be some kind of stop-light presentation that enforces the accounting.  In other words, if a maintenance event is "skipped", that needs to be evident.  So unless the reminder leads to a tracked maintenance event, it's really no better than the totally passive spreadsheet I'm using now.  There's lots of reasons for needing this functionality.  For example, if doing oil analysis, you have to keep a history of the intervals at which it is done and enforced as such.  Incidentally, is there a way to have a super-dashboard that shows all upcoming events chronologically regardless of the vehicle?  I have 11 old econoboxes I maintain as a home mechanic and tracking all of it is a nightmare with spreadsheets.

**Reply by hargata** [reply]
yeah…I don’t think there’s going to be any plans of enforcing accountability to scheduled maintenance and keeping a log of all missed maintenance. If the reminder is renewed then ideally there should be a service record that matches up to it when it was renewed, but it’s completely optional. 

You can always check the service record tab in retrospective to see if the oil changes follow a schedule or if it deviated from it.

as for a consolidated view of all reminders, there’s the kiosk view and the shop calendar view.

https://docs.lubelogger.com/Advanced/Kiosk
